lib/delta-sparql-mapping.js

```diff
@@ -0,0 +1,446 @@
+  await executeUpdate({
+    type: 'update',
+    updates: [
+      {
+        updateType: 'insert',
+        insert: [
+          {
+            type: 'graph',
+            triples: [parseSparqlJsonBindingQuad(triple)],
+            name: {
+              termType: 'NamedNode',
+              value: LANDING_ZONE_GRAPH,
+            },
+          },
+        ],
+      },
+    ],
+    prefixes: {},
+  });
+}
+
+async function deleteTripleInLandingZone(triple) {
+  await executeUpdate({
+    type: 'update',
+    updates: [
+      {
+        updateType: 'delete',
+        delete: [
+          {
+            type: 'graph',
+            triples: [parseSparqlJsonBindingQuad(triple)],
+            name: {
+              termType: 'NamedNode',
+              value: LANDING_ZONE_GRAPH,
+            },
+          },
+        ],
+      },
+    ],
+    prefixes: {},
+  });
+}
+
+async function executeUpdate(query, endpoint = TARGET_DATABASE_ENDPOINT) {
+  const queryStr =
+    typeof query === 'string' ? query : sparqlGenerator.stringify(query);
+
+  let attempts = 0;
+  while (attempts < MAX_DB_RETRY_ATTEMPTS) {
+    try {
+      await muUpdate(queryStr, {}, { sparqlEndpoint: endpoint });
+      return; // Exit the function if update is successful
+    } catch (e) {
+      attempts++;
+      if (attempts >= MAX_DB_RETRY_ATTEMPTS) {
+        console.log(
+          `Update failed after ${MAX_DB_RETRY_ATTEMPTS} retries: ${e.message} on ${endpoint}:`,
+        );
+        throw e; // Re-throw the error after max retries
+      }
+    }
+  }
+}
+
+async function executeSelect(query, endpoint = TARGET_DATABASE_ENDPOINT) {
+  let attempts = 0;
+  while (attempts < MAX_DB_RETRY_ATTEMPTS) {
+    try {
+      return await muQuery(
+        sparqlGenerator.stringify(query),
+        {},
+        { sparqlEndpoint: endpoint },
+      );
+    } catch (e) {
+      attempts++;
+      if (attempts >= MAX_DB_RETRY_ATTEMPTS) {
+        console.log(
+          `Query failed after ${MAX_DB_RETRY_ATTEMPTS} retries: ${
+            e.message
+          } on ${endpoint}:  
+          ${sparqlGenerator.stringify(query)}`,
+        );
+        throw e; // Re-throw the error after max retries
+      }
+    }
+  }
+}
+
+export async function initialMapping() {
+  const startTime = Date.now();
+  const insertQueries = await getInsertQueries();
+  for (const query of insertQueries) {
+    const queryAsString = sparqlGenerator.stringify(query);
+    console.log(
+      `Executing initial mapping query  (${insertQueries.indexOf(query) + 1} of ${insertQueries.length})`,
+    );
+    await executeUpdate(queryAsString, TRIPLE_STORE_ENDPOINT);
+    console.log(
+      `Successfully executed initial mapping query (${insertQueries.indexOf(query) + 1} of ${insertQueries.length})`,
+    );
+  }
+  const endTime = Date.now();
+  console.log(`Initial mapping took ${endTime - startTime}ms`);
+}
+
+export async function synchronizedDeltaProcessing(deltaMessage) {
+  await lock.acquire();
```

Comment:
I am not sure a lock is needed here.
Looking at the code in `delta-sync.js`, it indeed looks like it's possible for one delta-consumer job to start while another one is busy (and really doing something, not just due to a premature restart). And it makes sense not to want that. However, maybe I would fix it there, and not in the core logic of the `delta-sparql-mapping.js`. What do you think?